F-FrontEnd/src/F-compile-expr.c

```diff
@@ -742,9 +742,26 @@ compile_expression(expr x)
             doEmit:
 /* FEAST CHANGE start */
             /* if (type_is_not_fixed) */
-            if(type_is_not_fixed && tp)
+            if(type_is_not_fixed && tp) {
 /* FEAST CHANGE end */
+                // Save the TYPE_IS_FIXED attribute for left and right operand.
+                // Fix issue #207
+                int bLtype_fixed = TYPE_IS_FIXED(bLType);
+                int bRtype_fixed = TYPE_IS_FIXED(bRType);
+
+                // Apply the attribute NOT FIXED on the result type if one 
+                // operand type is NOT FIXED.
                 TYPE_SET_NOT_FIXED(bottom_type(tp));
+                
+                // Re-apply the attribute on the left operand
+                if(bLtype_fixed) {
+                    TYPE_UNSET_NOT_FIXED(bLType);
+                } 
+                // Re-apply the attribute on the right operand
+                if(bRtype_fixed) {
+                    TYPE_UNSET_NOT_FIXED(bRType);
```

Comment:
#207 seems to be occured if `bRType == bottom_type(tp)`.
So this line just reverts `TYPE_SET_NOT_FIXED(bottom_type(tp));` in line 754 and removes the `TYPE_IS_NOT_FIXED` attribute of `bottom_type(tp)`.

Reply:
@shingo-s So what would be your suggestion?

Reply:
I think `bottom_type(tp)` and `bRType` should be difference objects.
So replacing code between line 731 and line 734 like the following code will solve #207 .

```c
                if((biop == LOGIB || bType == bLType) && shape == lshape) {
                    tp = wrap_type(lt); // This `tp` turns into `bottom_type(tp)` later and `lt` == `bLType` if lt is not an array type
                } else if ((biop == LOGIB || bType == bRType) && shape == rshape) {
                    tp = wrap_type(rt); // same as above
```

Reply:
@shingo-s Ok, I'll do the change and check it. Thanks for the suggestion.